### PR TITLE
ci: update terraform for test-provisioned clusters

### PR DESCRIPTION
**Pull request by sdowell**
The e2e GKE clusters can now be provisioned using the test scaffolding. The CI jobs now use this functionality to create clusters, and no longer rely on pre-provisioned clusters. This also updates the dev environment to no longer create clusters, assuming that developers can use the test scaffolding to create clusters.

### CODE REVIEW

File: e2e/testinfra/terraform/prow/networks.tf [hunk 1]
```diff
@@ -27,3 +27,93 @@ resource "google_compute_subnetwork" "e2e-subnetwork-1" {
   description = "Subnetwork for use in e2e test clusters"
   private_ip_google_access = false
 }
+
+resource "google_compute_subnetwork" "e2e-subnetwork-2" {
```
Comment (tiffanny29631) [inline e2e/testinfra/terraform/prow/networks.tf:31]: How does the quota look like, do we have a limitation on the subnetworks

**Review by tiffanny29631 — COMMENTED** [review]
Have these clusters been created once? Given that the new CI is already up running.

File: e2e/testinfra/terraform/prow/networks.tf (hunk 1 above)
Comment (sdowell) [reply e2e/testinfra/terraform/prow/networks.tf:31]: All of these subnetworks fit within the quota, I've already ran this to set up the new test environment. I committed a bit of a faux pas by running this before getting it code reviewed, but I didn't want to have conflicting terraform definitions submitted on `main`.

**Comment by sdowell** [comment]
> Have these clusters been created once? Given that the new CI is already up running.

Yeah all of the clusters have already been created by the new prowjobs directly. They use the nets/subnets provisioned by this new terraform

File: e2e/testinfra/terraform/prow/networks.tf (hunk 1 above)
Comment (tiffanny29631) [reply e2e/testinfra/terraform/prow/networks.tf:31]: Got it. So in an ideal world the re-provision won't happen in a while? And the tf config aligns with the previous deployed setup?

Comment (sdowell) [reply e2e/testinfra/terraform/prow/networks.tf:31]: I didn't want to update these terraform configs until the migration was complete. The current configuration aligns with the legacy setup that is now being removed - my new changes reflect the new setup. I moved to a new project with the migration, so in the interim the previous project was still running with the legacy setup.

File: e2e/testinfra/terraform/README.md [hunk 2]
```diff
@@ -11,7 +11,6 @@ Follow the instructions to [install the Terraform CLI].
 
 # Dev Environments
 
-The Terraform config for dev environments is defined in the [dev directory].
```
Comment (karlkfi) [inline e2e/testinfra/terraform/README.md:14]: there's still tf config to set up the project and network, right?

File: e2e/testinfra/terraform/README.md [hunk 3]
```diff
@@ -90,4 +89,3 @@ terraform apply
 [command line variables]: https://www.terraform.io/language/values/variables#variables-on-the-command-line
 [install the Terraform CLI]: https://learn.hashicorp.com/tutorials/terraform/install-cli
 [prow directory]: ./prow
-[dev directory]: ./dev
```
Comment (karlkfi) [inline e2e/testinfra/terraform/README.md:93]: the dev dir still exits, right?

Comment (karlkfi) [reply e2e/testinfra/terraform/README.md:93]: oh, this is a named link, nvm.

File: e2e/testinfra/terraform/README.md (hunk 2 above)
Comment (karlkfi) [reply e2e/testinfra/terraform/README.md:14]: maybe add a section about setting up the env with tf?

Comment (sdowell) [reply e2e/testinfra/terraform/README.md:14]: Yes, there's just nothing specific in `dev/` anymore - as that was only used for setting up clusters before. Everything that is shared between dev and prow environment is in `common/` now, with some extra stuff for prow under `prow/`

Comment (sdowell) [reply e2e/testinfra/terraform/README.md:14]: This section should still contain accurate instructions for setting up a new dev and/or prow environment

File: e2e/testinfra/terraform/prow/networks.tf (hunk 1 above)
Comment (tiffanny29631) [reply e2e/testinfra/terraform/prow/networks.tf:31]: Understood, thanks for the clarification.

**Review by karlkfi — APPROVED** [review]
/lgtm

**Pull request merged**